src/main/java/vazkii/patchouli/api/QuadPredicate.java

```diff
@@ -0,0 +1,25 @@
+package vazkii.patchouli.api;
+
+import java.util.Objects;
+
+/**
+ * A predicate that takes four arguments and returns a boolean.
+ */
+@FunctionalInterface
+public interface QuadPredicate<T, U, V, W> {
+	boolean test(T t, U u, V v, W w);
```

Comment:
at this point just make this a specialized interface with concrete types instead of keeping it generic

Reply:
ok i'll change that

Reply:
changed it